### PR TITLE
util - helpers to play with endianness

### PR DESCRIPTION
**Pull request by ceolin**

### CODE REVIEW

File: src/shared/include/sol-util.h [hunk 1]
```diff
@@ -752,6 +752,161 @@ long int sol_util_strtol(const char *nptr, char **endptr, ssize_t len, int base)
 unsigned long int sol_util_strtoul(const char *nptr, char **endptr, ssize_t len, int base);
 
 /**
+ * @brief Swaps the bytes of a 16 bytes unsigned int
+ */
+#define uint16_bytes_swap(val) \
```
Comment (barbieri) [inline src/shared/include/sol-util.h]: if it goes in a header, namespace it with `sol_`

**Comment by vcgomes** [comment]
I would rename these functions to something like: `sol_util_get_{be,le}{16,32,64}(const void *ptr)` and `sol_util_put_{be,le}{16,32,64}(uint{16,32,64}, void *ptr)`

another alternative is use to the same nomeclature as the kernel:
`sol_util_cpu_to_be16(uint16)` and `sol_util_be16_to_cpu(uint16_t)`

for the conversions, this is a good reference:
http://commandcenter.blogspot.com/2012/04/byte-order-fallacy.html

File: src/shared/include/sol-util.h (hunk 1 above)
Comment (ceolin) [inline src/shared/include/sol-util.h]: doing it

**Comment by ceolin** [comment]
@vcgomes i'll follow the kernel nomeclature, but i'll also add sol_util_cpu_to_le16/sol_util_le16_to_cpu too

**Comment by ceolin** [comment]
https://github.com/solettaproject/soletta/pull/1515

**Pull request closed without merging**